### PR TITLE
Proof of work

### PR DESCRIPTION
**Pull request by 0xtlt**
Because the application must remain understandable to everyone, it will not be possible to choose its PoW level, instead 4 distinct levels will be available:

- Disabled: No filter and PoW on sent messages
- Level 1: PoW difficulty 5 (may change in the future)
- Level 2: PoW of difficulty 10 (may change in the future)
- Level 3: PoW of difficulty 15 (may change in the future)

It would be necessary to find a "naming" simpler than PoW on the other hand, like "security filter" or "anti spam filter"

----

- The "parent" messages of a message with a proof of work that is validated by the client will by default be validated without a check to avoid breaking the threads

### CODE REVIEW

File: damus/Views/ConfigView.swift [hunk 1]
```diff
@@ -7,6 +7,27 @@
...
 import SwiftUI
 
+enum ProofOfWorkPolicy: Int, CaseIterable {
+    case disabled = 0
+    case level1 = 5
+    case level2 = 10
+    case level3 = 25
+}
+
+func proofOfWorkPolicyText(_ policy: ProofOfWorkPolicy) -> String {
```
Comment (lionello) [inline damus/Views/ConfigView.swift:17]: This could be a member of the enum

Comment (0xtlt) [reply damus/Views/ConfigView.swift:17]: Will check tomorrow 👍

**Comment by jb55** [comment]
pow is already implemented in damus (count_hash_leading_zero_bits) it's just not used because it's not that useful for spam prevention at this time. all levels described here are extremely trivial to create on a single-core CPU in under a second.

**Comment by BenGWeeks** [comment]
My opinion is that Sats are the PoW that's already been done :-)

Shall we close this?

**Comment by jb55** [comment]
yeah PoW is a dead end imo

**Comment by 0xtlt** [comment]
Yep, I agree

I close this PR :)

**Pull request closed without merging**